src/BasicVault.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "../lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "../lib/yield-utils-v2/contracts/token/IERC2612.sol";
+
+
+/// @title Basic Vault
+/// @author hashedMae
+/// @notice A simple vault that tracks the amount of tokens a user has deposited and allows them to withdraw at any time
+/// @dev 
+
+contract BasicVault {
+
+IERC20 iToken;
+
+mapping(address => uint256) public userBalance;
+
+/// @notice Emitted when a user deposits token to the contract
+event Deposit(address indexed user, uint256 amount, uint256 balance);
+/// @notice Emitted when a user withdraws tokens from the contract
+event Withdrawal(address indexed user, uint256 amount, uint256 balance);
+
+/// @param iToken_ The address of the token that the vault will hold
+constructor(IERC20 iToken_){
+    iToken = iToken_;
+}
+
+/// @notice Allows a user to deposit tokens to the vault
+/// @dev Tracks user vault balance via mapping of address => uint
+/// @param amount Number of Tokens to deposit
+function depositToken(uint256 amount) public {
+    require(iToken.transferFrom(msg.sender, address(this), amount), "failed on tranfer");
+    userBalance[msg.sender] += amount;
```

Comment:
checks effects interactions

Reply:
This isn't as risky as the one in withdraw, but we still try to always stick to checks effects interactions pattern at all times.